python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1143,6 +1143,25 @@ def create_run(
         check.opt_mapping_param(run_config, "run_config", key_type=str)
         check.opt_str_param(mode, "mode")
 
+        check.opt_str_param(root_run_id, "root_run_id")
+        check.opt_str_param(parent_run_id, "parent_run_id")
+
+        # If step_keys_to_execute is None, then everything is executed.  In some cases callers
+        # are still exploding and sending the full list of step keys even though that is
+        # unnecessary.
```

Comment:
on large graphs this could be a non-trivial number of excess bytes stored in the run table? may be worth trying to resolve

Reply:
Good call. I will do that when consolidating into the single higher-level callsite